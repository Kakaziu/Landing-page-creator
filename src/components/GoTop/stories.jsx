import GoTop from '.';

export default {
  title: 'GoTop',
  component: GoTop,
};

export const Template = (args) => (
  <div style={{ height: '110vh' }}>
    <GoTop {...args} />
  </div>
);
