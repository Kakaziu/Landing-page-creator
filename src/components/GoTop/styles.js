import styled, { css } from 'styled-components';

export const Button = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 4rem;
    right: 4rem;
    width: 40px;
    height: 40px;
    border-radius: 20px;
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: rgba(10, 18, 40, 0.5);
    transition: all 0.3s ease-in-out;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.primaryColor};
    }
  `}
`;
