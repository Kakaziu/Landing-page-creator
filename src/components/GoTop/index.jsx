import { IoMdArrowDropup } from 'react-icons/io';
import * as Styled from './styles';

const GoTop = () => {
  return (
    <Styled.Button href="#" aria-label="Go to top" title="Go to top">
      <IoMdArrowDropup size="35" color="white" />
    </Styled.Button>
  );
};

export default GoTop;
