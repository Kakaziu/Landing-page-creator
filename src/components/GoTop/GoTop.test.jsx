import { render } from '@testing-library/react';
import GoTop from '.';

describe('<GoTop/>', () => {
  it('should you write something here', () => {
    render(<GoTop />);
  });
});
