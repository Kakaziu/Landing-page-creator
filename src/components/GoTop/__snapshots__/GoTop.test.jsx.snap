// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoTop/> should render a go to top button 1`] = `
.c0 {
  position: fixed;
  bottom: 4rem;
  right: 4rem;
  width: 40px;
  height: 40px;
  border-radius: 20px;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(10, 18, 40, 0.5);
  transition: all 0.3s ease-in-out;
  cursor: pointer;
}

.c0:hover {
  background-color: #0A1128;
}

<div>
  <a
    aria-label="Go to top"
    class="c0"
    href="#"
    title="Go to top"
  >
    <svg
      color="white"
      fill="currentColor"
      height="35"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 512 512"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128 320l128-128 128 128z"
      />
    </svg>
  </a>
</div>
`;
